google/datalab/contrib/pipeline/composer/_composer.py

```diff
@@ -29,22 +30,34 @@ def __init__(self, zone, environment):
     """
     self._zone = zone
     self._environment = environment
+    self._gcs_dag_location = None
 
   def deploy(self, name, dag_string):
     client = gcs.Client()
-    bucket = client.get_bucket(self.bucket_name)
-    filename = 'dags/{0}.py'.format(name)
+    try:
+      gcs_dag_location_splits = self.gcs_dag_location.split('/')
```

Comment:
_ , _, bucket, filepath = self.gcs_dag_location.split('/', 3)

You should have gcs_dag_location() verify that what it returns is in the expected format with trailing slash, etc.  And test whether this works with a top level directory in the bucket (i.e. gcs_dag_location = "gs://bucket/" )

Reply:
That made my life easy, thanks!